pkg/connection/settings.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connection
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// Flags defines Flags that may be bound to the Settings. Use like:
+// `cmd.PersistentFlags().AddFlagSet(connection.Flags)`
+var Flags *pflag.FlagSet
+
+var configPath string
+var activeConfigPointerFilename = "active_config"
+
+func init() {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		panic(err)
+	}
+	configPath = filepath.Join(home, ".config/registry")
+
+	Flags = pflag.NewFlagSet("registry", pflag.ExitOnError)
+	Flags.StringP("config", "c", "", "Name of a settings profile or path to config file")
+	Flags.String("registry.address", "", "the server and port of the registry api (eg. localhost:8080)")
+	Flags.Bool("registry.insecure", false, "set to true if client must connect via http (not https)")
```

Comment:
could we clarify this, possibly with "if specified, client connects via http (not https)"

Reply:
yup